examples/go/stream_benchmarker/benchmark/main.go

```diff
@@ -0,0 +1,417 @@
+	flag.StringVar(&filename, "o", filename, "Write report output to a file instead of standard out")
+
+	flag.Parse()
+
+	log.SetOutput(os.Stderr)
+	output := os.Stdout
+
+	if filename != "" {
+		file, err := os.Create(filename)
+		if err != nil {
+			log.Fatalf("Couldn't open output file. File: %v, Error: %v\n", filename, err)
+		}
+		defer file.Close()
+		output = file
+	}
+
+	satelliteStreamOptions := &benchmark.SatelliteStreamOptions{
+		AcceptedFraming: []stellarstation.Framing{stellarstation.Framing_AX25, stellarstation.Framing_BITSTREAM},
+		SatelliteID:     satelliteID,
+		APIKey:          apiKey,
+		Endpoint:        endpoint,
+	}
+
+	var streamChannel = make(chan *stellarstation.ReceiveTelemetryResponse)
+
+	ss, err := benchmark.OpenSatelliteStream(satelliteStreamOptions, streamChannel)
+	if err != nil {
+		log.Fatal(err)
+
+	}
+	defer ss.Close()
+
+	passEndedChan := make(chan bool)
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt)
+	reportingTicker := time.NewTicker(interval)
+	defer reportingTicker.Stop()
+
+	log.Println("Listening for messages")
+	var ps passSummary
+	var md metricsData
+	var passSummaries []passSummary
+	var passStarted = false
+	var done = false
+	var printedHeader = false
+	for !done {
+		select {
+		case streamResponse := <-streamChannel:
+			message := streamResponse.Telemetry
+			if len(message.Data) != 0 {
+				planID := streamResponse.PlanId
+				if passStarted == false {
+					if planID != "" {
+						log.Printf("Receiving messages for Plan ID %v", planID)
+						passStarted = true
+						firstTime, _ := ptypes.Timestamp(message.TimeFirstByteReceived)
+						md.initialTime = firstTime
+						ps.firstByteTime = firstTime
+						ps.initialPlanID = planID
+					}
+
+				}
+				timeFirstByteReceived, _ := ptypes.Timestamp(message.TimeFirstByteReceived)
+				timeLastByteReceived, _ := ptypes.Timestamp(message.TimeLastByteReceived)
+				numBytesInMessage := len(message.Data)
+
+				metric := metric{
+					timeFirstByteReceived: timeFirstByteReceived,
+					timeLastByteReceived:  timeLastByteReceived,
+					dataSize:              numBytesInMessage,
+					metricTime:            time.Now().UTC(),
+				}
+				md.metrics = append(md.metrics, metric)
+				if md.mostRecentTimeLastByteReceived.Before(timeLastByteReceived) {
+					md.mostRecentTimeLastByteReceived = timeLastByteReceived
+				}
+				md.planID = planID
+			}
+
+		case timeOfReporting := <-reportingTicker.C:
+			if passStarted {
+				md.timeOfReporting = timeOfReporting
+				outputDetails := md.compileDetails()
+				if !printedHeader {
+					outputDetails.printHeader(output)
+					printedHeader = true
+				}
+				if (assumeEndAfterDuration > 0) && (outputDetails.totalDataSize == 0) {
+					timeSincePassFirstByte := time.Now().UTC().Sub(ps.firstByteTime)
+					if timeSincePassFirstByte > assumeEndAfterDuration {
+						go func() { passEndedChan <- true }()
```

Comment:
If you want to fire a signal to tell the pass is ended, `close(passEndedChan)` is more natural way to do it.

Reply:
When this is left running for multiple passes,  we can't close the channel.

Reply:
I didn't recognize the use case, thanks for telling it. 
In that case how about moving the code in `case <-passEndedChan:` here? I guess passEndedChan doesn't have to be a channel since only this line sends true to the channel and it won't happen in parallel. Code to handle passEndedChan can be executed sequentially in this if statement, if I understand correctly, which can reduce an unexpected timing issue caused by using goroutine to send true to the channel.

Reply:
Yeah, good catch.